strawberryfields/ops.py

```diff
@@ -33,101 +33,97 @@
-register objects using the following syntax::
+register objects using the following syntax:
 
-  with eng:
-    G(params) | q
-    F(params) | (q[1], q[6], q[2])
+.. code-block:: python
 
-Here :samp:`eng` is an instance of :class:`strawberryfields.engine.Engine` and defines the context in which
-the commands are executed.
+  prog = sf.Program(3)
+  with prog.context as q:
+      G(params) | q
+      F(params) | (q[1], q[6], q[2])
+
+Here :samp:`prog` is an instance of :class:`strawberryfields.program.Program` and defines the context in which
+the commands are stored.
 Within each command, the part on the left is an :class:`Operation` instance,
 quite typically a constructor call for the requested operation class with the relevant parameters.
 The vertical bar calls the :func:`__or__` method of the :class:`Operation` object, with the part on the right as the parameter.
 The part on the right is a single :class:`strawberryfields.engine.RegRef` object or, for multi-mode gates, a sequence of them.
 It is of course also possible to construct gates separately and reuse them several times::
 
   R = Rgate(s)
-  with eng:
-    R   | q
-    Xgate(t) | q
-    R.H | q
+  with prog.context as q:
+      R   | q
+      Xgate(t) | q
+      R.H | q
 
 
 There are six kinds of :class:`Operation` objects:
 
 * :class:`Preparation` operations only manipulate the register state::
 
-    eng, q = sf.Engine(3)
-    with eng:
-      Vac | q[0]
-      All(Coherent(0.4, 0.2)) | (q[1], q[2])
+    with prog.context as q:
+        Vac | q[0]
+        All(Coherent(0.4, 0.2)) | (q[1], q[2])
 
 * Transformations such as :class:`Gate` and :class:`Channel` operations only manipulate the register state::
 
-    eng, q = sf.Engine(2)
-    with eng:
-      Dgate(0.3)   | q[0]
-      BSgate(-0.5) | q[0:2]
+    with prog.context as q:
+        Dgate(0.3)   | q[0]
+        BSgate(-0.5) | q[0:2]
 
 * :class:`Measurement` operations manipulate the register state and produce classical information.
-  The information is directly available only after the simulation has been run up to the point of measurement::
+  The information is directly available only after the program has been run up to the point of measurement::
+
+    with prog.context as q:
+        alice, bob = q
+        Measure       | alice
 
-    eng, (alice, bob) = sf.Engine(2)
-    with eng:
-      Measure       | alice
-      eng.run()
-      Dgate(alice.val) | bob
+    eng = sf.LocalEngine(backend='fock')
+    eng.run(prog)
+    print(alice.val)
 
   Alternatively one may use a symbolic reference to the register containing the measurement result
   by supplying registers as the argument to an :class:`Operation`, in which case the measurement may be deferred,
   i.e., we may symbolically use the measurement result before it exists::
 
-    eng, (alice, bob) = sf.Engine(2)
-    with eng:
-      Measure   | alice
-      Dgate(alice) | bob
-    eng.run()
+    with prog.context as q:
+        alice, bob = q
```

Comment:
Due to tuple unpacking, would
```python
with prog.context as alice, bob:
```
also work?

Reply:
No, but `with prog.context as (alice, bob):` does! That's nicer, I'll change it.